### PR TITLE
[Fix] Fetch Serial No by default for  Delivery Note Item

### LINKED ISSUE #6706: Fix Serial Number by default FIFO for Delivery Note

### PR DESCRIPTION
**Pull request by KanchanChauhan**
Fixed https://github.com/frappe/erpnext/issues/6706

### CODE REVIEW

File: erpnext/stock/doctype/delivery_note/delivery_note.js [hunk 1]
```diff
@@ -288,3 +288,38 @@ if (sys_defaults.auto_accounting_for_stock) {
 		}
 	}
 }
+
+frappe.ui.form.on('Delivery Note', {
+	set_serial_no: function(doc, cdt, cdn) {
+		var d = frappe.model.get_doc(cdt, cdn);
+		d.qty||(d.qty=1)
```
Comment (nabinhait) [inline erpnext/stock/doctype/delivery_note/delivery_note.js]: ???

File: erpnext/stock/doctype/delivery_note/delivery_note.js [hunk 2]
```diff
@@ -288,3 +288,38 @@ if (sys_defaults.auto_accounting_for_stock) {
...
+		if(!d.item_code && !d.warehouse && !d.qty) return;
```
Comment (nabinhait) [inline erpnext/stock/doctype/delivery_note/delivery_note.js]: all the 3 fields are mandatory right? In that case, condition should be based on "OR"

File: erpnext/stock/doctype/delivery_note/delivery_note.js [hunk 3]
```diff
@@ -288,3 +288,38 @@ if (sys_defaults.auto_accounting_for_stock) {
 		}
 	}
 }
+
+frappe.ui.form.on('Delivery Note', {
+	set_serial_no: function(doc, cdt, cdn) {
```
Comment (nabinhait) [inline erpnext/stock/doctype/delivery_note/delivery_note.js]: use `frm` instead of `doc` in args

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by KanchanChauhan** [comment]
Changes done

**Comment by KanchanChauhan** [comment]
Call set_serial_no from server side

**Review by nabinhait — APPROVED** [review]

**Pull request merged**